scripts/safari.js

```diff
@@ -336,46 +338,86 @@ function Safari() {
             "name" : "Forest",
             "types" : ["Grass", "Bug"],
             "excludeTypes" : [],
-            "include" : [16, 17, 18, 25, 163, 164],
-            "exclude" : [492, 649],
-            "customBST" : {"289" : 600},
-            "maxBST" : 600,
-            "minBST" : 300
+            "include" : [16, 17, 18, 25, 163, 164, 438, 185, 452, 65948, 131484, 65949, 131485, 649, 172, 287, 288, 289],
+            "exclude" : [649],
+            "customBST" : {
+                "289" : 600
+            },
+            "minBST" : 300,
+            "maxBST" : 601
         },
-        "river" : {
-            "name" : "River",
+        "sea" : {
+            "name" : "Sea",
             "types" : ["Water"],
-            "excludeTypes" : ["Ice"],
-            "include" : [524954, 656026, 721562, 852634],
-            "exclude" : [245]
+            "excludeTypes" : [],
+            "include" : [524954, 656026, 721562, 852634, 399, 489, 65958, 65959, 131423, 298, 712, 713, 686, 687, 691, 249, 382],
+            "exclude" : [721],
+            "customBST" : {
+                "249" : 630,
+                "382" : 630
+            },
+            "minBST" : 300,
+            "maxBST" : 631
         },
         "volcano" : {
             "name" : "Volcano",
             "types" : ["Fire", "Rock"],
             "excludeTypes" : ["Water", "Ice"],
-            "include" : [1049242],
-            "exclude" : []
+            "include" : [1049242, 721, 208],
+            "exclude" : [438, 185, 639, 345, 346],
+            "customBST" : {},
+            "minBST" : 315,
+            "maxBST" : 601
         },
         "cave" : {
             "name" : "Cave",
             "types" : ["Rock", "Ground", "Dark"],
             "excludeTypes" : ["Flying"],
-            "include" : [41, 42, 169, 92, 93, 202, 360],
-            "exclude" : []
+            "include" : [41, 42, 169, 92, 93, 202, 360, 486, 29, 30, 32, 33, 206],
+            "exclude" : [],
+            "customBST" : {
+                "486" : 600
+            },
+            "minBST" : 300,
+            "maxBST" : 601
         },
         "sky" : {
             "name" : "Sky",
             "types" : ["Flying"],
             "excludeTypes" : ["Bug"],
-            "include" : [329, 330, 635],
-            "exclude" : [328159]
+            "include" : [329, 330, 635, 380, 381, 250, 621],
+            "exclude" : [6, 130],
+            "customBST" : {
+                "250" : 620
```

Comment:
Rotom is not in. Alternate forms only get included if added manually, they are not added by type.
I left Lugia out because they are more related to the sea than the sky.